ding_unilogin/src/OAuth2/Client/Provider/StilUniloginProvider.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Drupal\ding_unilogin\OAuth2\Client\Provider;
+
+use Drupal\ding_unilogin\Exception\StilUniloginException;
+use GuzzleHttp\Exception\ConnectException;
+use League\OAuth2\Client\Provider\GenericProvider;
+use League\OAuth2\Client\Token\AccessToken;
+
+/**
+ * Stil Unilogin provider.
+ *
+ * @see https://viden.stil.dk/display/OFFSKOLELOGIN/OIDC
+ */
+class StilUniloginProvider extends GenericProvider {
+
+  /**
+   * The token introspect URL.
+   *
+   * @var string
+   */
+  protected $urlTokenIntrospect;
+
+  /**
+   * The end session URL.
+   *
+   * @var string
+   */
+  protected $urlEndSession;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getRequiredOptions() {
+    return array_merge(
```

Comment:
These array_merges could have been written as below and the array could have been a class variable, as it's use both here and in the next function.
```
return parent::getRequiredOptions() +
      [
        'urlTokenIntrospect',
        'urlEndSession',
      ];
```

Reply:
`+` will not work with lists (non-associative arrays), but introducing a private class variable makes sense. Done in https://github.com/eReolen/ereolengo/pull/105/commits/0b6c0d83576dc2608f97972cd7bad83e543b96b3.